app/src/main/java/com/lastaoutdoor/lasta/ui/screen/discover/FilterScreen.kt

```diff
@@ -19,190 +21,207 @@ import androidx.compose.material3.ExperimentalMaterial3Api
-              }
-            })
-
-        Spacer(modifier = Modifier.height(16.dp))
-
-        // Filter by activity type
-        Text(
-            text = stringResource(id = R.string.filter_activity_type),
-            style = TextStyle(fontSize = 20.sp, lineHeight = 24.sp, fontWeight = FontWeight(500)))
-        Row(verticalAlignment = Alignment.CenterVertically) {
-          activities.forEachIndexed { index, activityType ->
-            RadioButton(selected = index == selectedIndex, onClick = { selectedIndex = index })
-            Text(text = activityType.toString())
+  Column(modifier = Modifier.fillMaxSize().testTag("filterScreen")) {
+    MediumTopAppBar(
+        title = {
+          Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Center) {
+            Text(
+                text = stringResource(id = R.string.filter_options),
+                style = MaterialTheme.typography.titleLarge,
+                color = MaterialTheme.colorScheme.onBackground)
           }
-        }
-
-        Spacer(modifier = Modifier.height(4.dp))
-        HorizontalDivider(thickness = 2.dp, color = PrimaryBlue)
-        Spacer(modifier = Modifier.height(16.dp))
-
-        // Filter by difficulty level
-        Text(
-            text = stringResource(id = R.string.filter_difficulty_level),
-            style = TextStyle(fontSize = 20.sp, lineHeight = 24.sp, fontWeight = FontWeight(500)))
-        Row(verticalAlignment = Alignment.CenterVertically) {
-          UserLevel.values().forEach { levelItem ->
-            RadioButton(
-                selected = activitiesLevelArray[selectedIndex] == levelItem,
-                onClick = {
-                  activitiesLevelArray[selectedIndex] = levelItem
-                  setSelectedLevels(
-                      UserActivitiesLevel(
-                          climbingLevel = activitiesLevelArray[0],
-                          hikingLevel = activitiesLevelArray[1],
-                          bikingLevel = activitiesLevelArray[2]))
-                })
-            Text(text = userLevelToDifficultyLevel(levelItem))
+        },
+        navigationIcon = {
+          IconButton(onClick = { navigateBack() }) {
+            Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = null)
+          }
+        })
+
+    HorizontalDivider(modifier = Modifier.fillMaxWidth())
+
+    Spacer(modifier = Modifier.height(16.dp))
+
+    // Filter by activity type
+    Column(
+        modifier = Modifier.fillMaxSize().padding(16.dp),
+        horizontalAlignment = Alignment.Start,
+        verticalArrangement = Arrangement.SpaceBetween) {
+          Column {
+            Text(
+                text = stringResource(id = R.string.filter_activity_type),
+                style = MaterialTheme.typography.headlineSmall,
+                fontWeight = FontWeight.Bold)
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            FlowRow(
```

Comment:
I don't think that the button in the FlowRow are centered /spaced arround correctly. If you look at the vidéo preview, there is more space from the last button to the screen border than from the left screen border to the first button

Reply:
Yes indeed. I felt that arranging items from a FlowRow harder than usual, as it's constly adapting with n or fewer elements, hence spacing can sometimes be awkward depending on the size of the buttons